### PR TITLE
fix(select): adjust wrong type for `rootVariantClass` property

### PR DESCRIPTION
**Pull request by whiplashwebb**
## Proposed Changes

- Fixes a mistake I made in the types for select. rootVariantClass should be a ClassDefinition not a boolean. We regret the error.
- Turns out I should have been generating types rather than writing them. This Pr has been a learning experience. I regenerated the types and then patched the output to deal with a possible bug where the types for NotificationNotice were missing.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,8 @@ packages/docs/.vitepress/.cache/
 packages/docs/.vitepress/.temp/
 packages/docs/.vitepress/dist/
 packages/oruga/README.md
+packages/docs-next/.vitepress/cache/
```
Comment (mlmoravek) [inline .gitignore:13]: the package name got changed from docs-next to docs, as well as oruga-next to next. So no need for this adding.

File: packages/oruga/src/components/types.ts [hunk 2]
```diff
@@ -865,7 +865,7 @@ but will set body to position fixed, might break some layouts. */
                 /** Class of the select arrow */
                 arrowClass: ClassDefinition;
                 /** Class of the select variant for the root element */
-                rootVariantClass: boolean;
+                rootVariantClass: ClassDefinition;
```
Comment (mlmoravek) [inline packages/oruga/src/components/types.ts:868]: did you genereate the file?

File: .gitignore (hunk 1 above)
Comment (whiplashwebb) [reply .gitignore:13]: Clearly I need more coffee. You're right, i just had leftover files in the old directories locally.

File: packages/oruga/src/components/types.ts (hunk 2 above)
Comment (whiplashwebb) [reply packages/oruga/src/components/types.ts:868]: oh no... i've been manually editing a generated file... 🤦

**Pull request closed without merging**